src/main/scala/com/typesafe/sbt/bundle/SbtBundle.scala

```diff
@@ -261,70 +284,96 @@ object SbtBundle extends AutoPlugin {
   }
 
-  private def getConfig: Def.Initialize[Task[String]] = Def.task {
-    val checkInitialDelayInSeconds =
-      if (checkInitialDelay.value.toMillis % 1000 > 0)
-        checkInitialDelay.value.toSeconds + 1 // always round up
-      else
-        checkInitialDelay.value.toSeconds
-    val checkComponents = if (checks.value.nonEmpty)
-      checks.value.map(uri => s""""$uri"""").mkString(
-        s""",
-           |  "${(normalizedName in Bundle).value}-status" = {
-           |    description      = "Status check for the bundle component"
-           |    file-system-type = "universal"
-           |    start-command    = ["check", "--initial-delay", "$checkInitialDelayInSeconds", """.stripMargin,
-        ", ",
-        s"""]
-           |    endpoints        = {}
-           |  }""".stripMargin)
-    else
-      ""
-
-    s"""|version              = "1.1.0"
-        |name                 = "${(normalizedName in Bundle).value}"
-        |compatibilityVersion = "${compatibilityVersion.value}"
-        |system               = "${system.value}"
-        |systemVersion        = "${systemVersion.value}"
-        |nrOfCpus             = ${nrOfCpus.value}
-        |memory               = ${memory.value.underlying}
-        |diskSpace            = ${diskSpace.value.underlying}
-        |roles                = ${formatSeq(roles.value)}
-        |components = {
-        |  "${(normalizedName in Bundle).value}" = {
-        |    description      = "${projectInfo.value.description}"
-        |    file-system-type = "${bundleType.value}"
-        |    start-command    = ${formatSeq(startCommand.value)}
-        |    endpoints        = ${formatEndpoints(endpoints.value)}
-        |  }$checkComponents
-        |}
-        |""".stripMargin
+  private def getConfig(config: Configuration, filter: ScopeFilter): Def.Initialize[Task[String]] = Def.task {
+    val checkComponents = checks.?.all(filter).value.flatMap {
+      case Some(checksValue) if checksValue.nonEmpty =>
+        val checkInitialDelayValue = (checkInitialDelay in config).value
+        val checkInitialDelayInSeconds =
+          if (checkInitialDelayValue.toMillis % 1000 > 0)
+            checkInitialDelayValue.toSeconds + 1 // always round up
+          else
+            checkInitialDelayValue.toSeconds
+        checksValue.map(uri => s""""$uri"""").mkString(
+          s"""components."${(normalizedName in config).value}-status" = {
+             |  description      = "Status check for the bundle component"
+             |  file-system-type = "universal"
+             |  start-command    = ["check", "--initial-delay", "$checkInitialDelayInSeconds", """.stripMargin,
+          ", ",
+          s"""]
+             |  endpoints        = {}
+             |}""".stripMargin)
+      case _ =>
+        Seq.empty[String]
+    }
+
+    val declarations =
+      normalizedName.?.all(filter).value.collect {
+        case Some(value) => s"""name = "$value""""
+      } ++
```

Comment:
Just to provide the full picture, I want users to be able to express bundle configurations given a custom sbt configuration. Here's an example approach from that I'm playing with for the ReactiveMaps project:

```scala
// Main bundle configuration

normalizedName in Bundle := "reactive-maps-frontend"
BundleKeys.nrOfCpus in Bundle := 2.0
BundleKeys.memory in Bundle := 64.MiB
BundleKeys.diskSpace in Bundle := 50.MB
BundleKeys.endpoints in Bundle := Map(
  "akka-remote" -> Endpoint("tcp"),
  "web" -> Endpoint("http", services=Set(URI("http://:9000")))
)
BundleKeys.roles in Bundle := Set("dmz")
BundleKeys.startCommand in Bundle ++= Seq(
  "-Dhttp.address=$WEB_BIND_IP",
  "-Dhttp.port=$WEB_BIND_PORT",
  "-Dakka.cluster.roles.1=frontend"
)

// Configuration bundles that override the main one

lazy val BackendRegion = config("backend-region").extend(BundleConfiguration)
SbtBundle.configurationSettings(BackendRegion)
inConfig(BackendRegion)(Seq(
  normalizedName := "reactive-maps-backend-region",
  BundleKeys.configurationName := (normalizedName in BackendRegion).value,
  BundleKeys.endpoints := Map("akka-remote" -> Endpoint("tcp")),
  BundleKeys.roles := Set("intranet"),
  BundleKeys.startCommand :=
    Seq((BundleKeys.executableScriptPath in BackendRegion).value) ++
      (javaOptions in BackendRegion).value ++
      Seq(
        "-Dakka.cluster.roles.1=backend-region",
        "-main", "backend.Main"
      )
))

lazy val BackendSummary = config("backend-summary").extend(BackendRegion)
SbtBundle.configurationSettings(BackendSummary)
inConfig(BackendSummary)(Seq(
  normalizedName := "reactive-maps-backend-summary",
  BundleKeys.configurationName :=  (normalizedName in BackendSummary).value,
  BundleKeys.startCommand :=
    Seq((BundleKeys.executableScriptPath in BackendRegion).value) ++
      (javaOptions in BackendSummary).value ++
      Seq(
        "-Dakka.cluster.roles.1=backend-summary",
        "-main", "backend.Main"
      )
))
```

So, there's a main bundle configuration, and then a new one within its own configuration named `BackendRegion`. `BackendSummary` extends `BackendRegion`. FYI ReactiveMaps is one app that can be run in 3 ways (frontend, backend-region and backend-summary), thus resulting in 3 bundles.

When I type `bundle:dist` I'll get the main bundle. When I type `bundle-region:dist` or `bundle-summary:dist` I want optional configuration created i.e `bundle.conf` files containing *just* those properties that have been declared for that configuration.

HTH.